test/api/reduce.test.ts

```diff
@@ -25,8 +28,7 @@ describe("reduce", () => {
   test("it can produce a result of a different type", async () => {
     const result = await pipe(
       CB.fromIter([1, 2, 3, 4, 5]),
-      // TODO: Make it so explicit type params are not needed
-      CB.reduce<number, string>((prev, curr) => `${prev}${curr}`),
+      CB.reduce((prev, curr) => `${prev}${curr}`, ""),
```

Comment:
All though TypeScript can't infer the types correctly
without an initial value, the code still works as expected.

I guess this is the most allegiant approach (just requiring an initial value when changing the return type).